frontend/src/pages/onboarding/index.tsx

```diff
@@ -47,15 +55,19 @@ const onboardingSteps: OnboardingStep[] = [
     subtitle: 'Other climbers will use this to identify you',
     form: <UsernameForm />,
     validate: {
-      name: Yup.string().required('Name is required.'),
+      name: Yup.string()
+        .min(MIN_NAME_LEN, NAME_LEN_ERROR)
+        .max(MAX_NAME_LEN, NAME_LEN_ERROR)
+        .matches(REGEX_NAME, NAME_REGEX_ERROR)
+        .required('Name is required.'),
     },
   },
   {
     title: 'Complete your profile',
     subtitle: 'Help other climbers know more about you',
     form: <DetailsForm />,
     validate: {
-      height: Yup.number().positive().integer().max(300).optional(),
+      height: Yup.number().positive().integer().max(MAX_HEIGHT).optional(),
       reach: Yup.number().positive().integer().optional(),
```

Comment:
Can consider changing Reach to be a selector from -20cm to +20cm (@therizhao confirmed that these are the lower and upper bounds for reach). Not sure if there's a better way to implement this